### DISCUSSION TITLE
Catch-all broadcast authorization callbacks

### DISCUSSION THREAD

**Discussion by sebastiandedeyne** [laravel/framework#46325] [Q&A] [answered]
We used to have the following authorization callback defined in in `routes/channels.php`:

```php
Broadcast::channel('{key}', function (User $user) {
    if (! $user->isAdmin()) {
        return false;
    }

    return [
        'id' => $user->id,
        'name' => $user->name,
    ];
});
```

The `{key}` parameter was able to act as a catch-all route for all channels of the application so we didn't need to specify them individually.

Back in October, Taylor merged a [PR](https://github.com/laravel/framework/pull/44531) that made the matching `Broadcast::channel` more robust. I agree with the changes made, however it breaks our setup for a catch-all route.

Are there any alternatives to achieve the same without extending overriding the `BroadCaster` class?

I'd love to be able to define a default auth check without having to override a class from vendor.

**Comment by franck-grenier** [comment]
Hello @sebastiandedeyne ,
we have the same need of a "catch-all" channel.

Did you find a solution ?

**Comment by Angel5a** [comment, marked as answer]
@sebastiandedeyne I can't find a proper preg escaping in the Broadcaster implementation, so try to use '{key}.*' as a channel name. Would it work - fill free to fill the bug report.

**Reply by LukeAbell** [reply]
Thanks @Angel5a! That worked!
